crates/forge/src/main.rs

```diff
@@ -14,3 +16,83 @@ fn main() {
         }
     };
 }
+
+fn init_logging() -> Option<impl Drop> {
+    use chrono::Local;
+    use std::fs;
+
+    use std::path::PathBuf;
+    use tracing_chrome::ChromeLayerBuilder;
+    use tracing_subscriber::filter::{EnvFilter, LevelFilter, Targets};
+    use tracing_subscriber::fmt::Layer;
+    use tracing_subscriber::fmt::time::Uptime;
+    use tracing_subscriber::prelude::*;
+
+    let mut guard = None;
+
+    let fmt_layer = Layer::new()
+        .with_writer(io::stderr)
+        .with_ansi(io::stderr().is_terminal())
+        .with_timer(Uptime::default())
+        .with_filter(
+            EnvFilter::builder()
+                .with_default_directive(LevelFilter::WARN.into())
+                .with_env_var("SNFORGE_LOG")
+                .from_env_lossy(),
+        );
+
+    let tracing_profile = env::var("SNFORGE_TRACING_PROFILE").ok().is_some_and(|var| {
+        let s = var.as_str();
+        s == "true" || s == "1"
+    });
+
+    let profile_layer = if tracing_profile {
+        let mut path = PathBuf::from(format!(
+            "./snforge-profile-{}.json",
+            Local::now().to_rfc3339()
+        ));
+
+        // Create the file now, so that we early panic, and `fs::canonicalize` will work.
+        let profile_file = fs::File::create(&path).expect("failed to create profile file");
+
+        // Try to canonicalize the path so that it is easier to find the file from logs.
+        if let Ok(canonical) = fs::canonicalize(&path) {
+            path = canonical;
+        }
+
+        eprintln!(
+            "`snforge` run will output tracing profile to: {}",
+            path.display()
+        );
+        eprintln!(
+            "Open that file with https://ui.perfetto.dev (or chrome://tracing) to analyze it"
```

Comment:
nit: Could we provide a ready-to-open link for Chrome? Could be handy

Reply:
I think `ui.perfetto.dev` is better anyway, don't think the link is that useful.